ocaml/xapi/xapi_pif_helpers.ml

```diff
@@ -177,15 +177,20 @@ let assert_tunnel_is_allowed_on_pif ~__context ~transport_PIF =
   | _ ->
       ()
 
-let tunnel_is_allowed_on_pif ~__context ~transport_PIF =
+let tunnel_of_private_network_is_allowed_on_pif ~__context ~transport_PIF =
   let pif_rec = Db.PIF.get_record ~__context ~self:transport_PIF in
   match get_pif_topo ~__context ~pif_rec with
   | Physical pif_rec :: _ when pif_rec.API.pIF_bond_slave_of <> Ref.null ->
       (* Disallow creating on bond slave *)
       (* Here we rely on the implementation to guarantee that `Physical` is a terminating case *)
       false
-  | Tunnel_access _ :: _ ->
-      false
+  | Tunnel_access tunnel_ref :: _ -> (
```

Comment:
I think the logic is not correct here: `Tunnel_access` is the pif created by `xapi_tunnel`, it should never be allowed to setup tunnel on it again.